store/src/pmmr.rs

```diff
@@ -138,19 +138,27 @@ where
 		if self.rm_log.includes(position) {
 			return None;
 		}
+
 		// ... or in the prune list
-		let prune_shift = match self.pruned_nodes.get_leaf_shift(position) {
-			Some(shift) => shift,
-			None => return None,
-		};
+		// let prune_shift = match self.pruned_nodes.get_leaf_shift(position) {
+		// 	Some(shift) => shift,
+		// 	None => return None,
+		// };
 
 		let hash_val = self.get_from_file(position);
 		if !include_data {
 			return hash_val.map(|hash| (hash, None));
 		}
 
+		// if this is not a leaf then we have no data
+		if !pmmr::is_leaf(position) {
+			return hash_val.map(|hash| (hash, None));
+		}
+
 		// Optionally read flatfile storage to get data element
```

Comment:
Data file compaction commented out for now - so shift is always 0 for now.
Will tackle in separate PR as discussed above.